src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -1140,9 +1141,9 @@ int ciMethod::instructions_size() {
 // ciMethod::log_nmethod_identity
 void ciMethod::log_nmethod_identity(xmlStream* log) {
   GUARDED_VM_ENTRY(
-    CompiledMethod* code = get_Method()->code();
-    if (code != NULL) {
-      code->log_identity(log);
+    CodeBlob* code = get_Method()->code();
+    if (code != NULL && code->is_compiled()) {
+      code->as_compiled_method()->log_identity(log);
```

Comment:
Doesn't this change the log output?

Reply:
Yes, the output of MH intrinsic cases will become like the cases where codes are not set. The removed parts are compiled ID, compiler, and compile level. They are all constants for MH intrinsics, not providing much information.